openapi.yaml

```diff
@@ -16,175 +16,218 @@ host: api.oregonstate.edu
+          in: query
+          type: string
+          required: true
+          description: 9-digit OSU ID of student whose notes will be retrieved
+        - name: q
+          in: query
+          type: string
           required: false
+          description: Query string to be used when searching through the contents of each note
+        - name: sortKey
+          in: query
           type: string
+          enum:
+            - 'lastModified'
+            - 'source'
+            - 'permissions'
+            - 'contextType'
+          default: 'lastModified'
+          required: false
+          description: The key by which to sort the notes in the response
+        - name: sources
+          in: query
+          type: array
+          items:
+            type: string
+            enum: &SOURCES
+              - advisorPortal
+              - banner
+              - degreeWorks
+            collectionFormat: csv
+          required: false
+          description: A comma-separated list of sources to filter by
+        - name: contextTypes
+          in: query
+          type: array
+          items:
+            type: string
+            collectionFormat: csv
+          required: false
+          description: A comma-separated list of context types to filter by
       responses:
         200:
           description: Successful response
           schema:
-            $ref: '#/definitions/PetsResult'
+            $ref: '#/definitions/NotesResult'
         400:
-          description: Bad Request
+          description: Bad request
           schema:
             $ref: '#/definitions/Errors'
         500:
-          description: Internal Server Error
+          description: Internal server error
           schema:
             $ref: '#/definitions/Errors'
-  /pets/{id}:
-    get:
-      summary: Info for a specific pet
+    post:
+      summary: Create a new note
       tags:
-        - pets
-      description: Get pet by unique ID
-      operationId: getPetById
+        - notes
+      operationId: postNotes
       parameters:
-        - $ref: '#/parameters/authorization'
-        - name: id
-          in: path
-          description: Unique Pet ID
-          required: true
-          type: string
+        - in: body
+          name: requestBody
+          schema:
+            type: object
+            properties:
+              note:
+                $ref: '#/definitions/note'
+              studentID:
+                $ref: '#/definitions/studentID'
+              creatorID:
+                $ref: '#/definitions/creatorID'
+              permissions:
+                $ref: '#/definitions/permissions'
+              context:
+                $ref: '#/definitions/context'
+            required:
+              - note
+              - studentID
+              - creatorID
+              - context
+      responses:
+        201:
+          description: Successfully created a note
+          schema:
+            $ref: '#/definitions/NoteResult'
```

Comment:
With this [response](https://jsonapi.org/format/#crud-creating-responses-201), we should also return `Location` in the response header:

> The response SHOULD include a Location header identifying the location of the newly created resource.

We can also define a response header with OpenAPI: https://swagger.io/docs/specification/2-0/describing-responses/

Reply:
Fixed